test/e2e/apiextensions_test.go

```diff
@@ -137,3 +138,74 @@ func TestComposition(t *testing.T) {
 			Setup(setup).Feature(),
 	)
 }
+
+func TestValidation(t *testing.T) {
+
+	manifests := "test/e2e/manifests/apiextensions/validation/composition-schema"
+	composition := features.Table{
+		{
+			Name: "PrerequisitesAreCreated",
+			Assessment: funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "prerequisites/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "prerequisites/*.yaml"),
+			),
+		},
+		{
+			Name:       "XRDBecomesEstablished",
+			Assessment: funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "prerequisites/definition.yaml", apiextensionsv1.WatchingComposite()),
+		},
+		{},
+		{
+			Name: "ClaimIsCreated",
+			Assessment: funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "claim.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "claim.yaml"),
+			),
+		},
+		{
+			Name:       "ClaimBecomesAvailable",
+			Assessment: funcs.ResourcesHaveConditionWithin(2*time.Minute, manifests, "claim.yaml", xpv1.Available()),
+		},
+		{
+			Name:       "ClaimHasPatchedField",
+			Assessment: funcs.ResourcesHaveFieldValueWithin(2*time.Minute, manifests, "claim.yaml", "status.coolerField", "I'M COOL!"),
+		},
+		{
+			Name: "ClaimIsDeleted",
+			Assessment: funcs.AllOf(
+				funcs.DeleteResources(manifests, "claim.yaml"),
+				funcs.ResourcesDeletedWithin(2*time.Minute, manifests, "claim.yaml"),
+			),
+		},
+		{
+			Name: "PrerequisitesAreDeleted",
+			Assessment: funcs.AllOf(
+				funcs.DeleteResources(manifests, "prerequisites/*.yaml"),
+				funcs.ResourcesDeletedWithin(3*time.Minute, manifests, "prerequisites/*.yaml"),
+			),
+		},
+	}
+
+	// Enable our feature flag.
+	setup := funcs.AllOf(
+		funcs.AsFeaturesFunc(funcs.HelmUpgrade(HelmOptions(helm.WithArgs("--set args={--debug,--enable-composition-webhook-schema-validation}"))...)),
+		funcs.ReadyToTestWithin(1*time.Minute, namespace),
+	)
+
+	// Disable our feature flag.
+	teardown := funcs.AllOf(
+		funcs.AsFeaturesFunc(funcs.HelmUpgrade(HelmOptions()...)),
+		funcs.ReadyToTestWithin(1*time.Minute, namespace),
+	)
```

Comment:
I wonder whether it would make sense to have every test "upgrade" the Crossplane deployment to have whatever flags they need. Kinda weird, but it could be more resilient than relying on tests that change how Crossplane is installed to remember to undo their changes.

Reply:
I'm going to keep this as is - we can change if/when we find a better pattern.